libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -168,10 +168,25 @@ bool AP_Arming::barometer_checks(bool report)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_BARO)) {
-        if (!AP::baro().all_healthy()) {
+        const AP_Baro *baro = AP_Baro::get_singleton();
```

Comment:
Same as the GPS comment below, store a reference rather then a pointer, which would need to be null pointer checked.

Reply:
Done